app/src/docs/v1.api-spec.yaml

```diff
@@ -194,6 +199,32 @@ paths:
           $ref: "#/components/responses/Forbidden"
         default:
           $ref: "#/components/responses/Error"
+  /bucket/{bucketId}/sync:
+    get:
+      summary: Synchronize a bucket
+      description: |-
+        Synchronize the contents of an existing bucket with COMS so that it can track and manage the objects
+        in it. This avoids the need to re-upload preexisting files through the COMS API.
+
+        It looks at all objects in the bucket and adds them to an internal processing queue;
+        once it is eventually processed, it is tracked by COMS.
+      tags:
+        - Sync
+      operationId: syncBucket
+      parameters:
+        - $ref: "#/components/parameters/Path-BucketId"
+        - $ref: "#/components/parameters/Query-Public"
```

Comment:
This is inaccurate - the query parameter for this is `full`, not `public`. You'll need to make a new Query-Full component parameter.

Reply:
Did you mean `all` instead of `full`? This is the closest thing I see in `sync.js`:

https://github.com/bcgov/common-object-management-service/blob/4473d8a848f1c9c8a5fdcde7e24faddab087d847/app/src/controllers/sync.js#L23

That being said, now that you mention it, I see `req.query.full` in `syncObject` - is that something that needs to be edited as well?

https://github.com/bcgov/common-object-management-service/blob/4473d8a848f1c9c8a5fdcde7e24faddab087d847/app/src/controllers/sync.js#L65

Reply:
It should be full. Will fix in upcoming PR.